pkg/cmd/experimental/syncgroups/ad/ldapinterface.go

```diff
@@ -94,13 +94,13 @@ func (e *ADLDAPInterface) ListGroups() ([]string, error) {
 		return nil, err
 	}
 
-	return sets.KeySet(reflect.ValueOf(e.ldapGroupToLDAPMembers)).List(), nil
+	return sets.KeySet(reflect.ValueOf(e.cachedGroupedUsers)).List(), nil
 }
 
 // populateCache queries all users to build a map of all the groups.  If the cache has already been
 // populated, this is a no-op.
 func (e *ADLDAPInterface) populateCache() error {
-	if e.cachePopulated {
+	if len(e.cachedGroupedUsers) > 0 {
```

Comment:
> If you consider what must happen, however, for this bug to manifest, you'll see it should be non-issue unless there's an intern or something building their LDAP syncer incorrectly. In order to ExtractMembers you need a UID, and to get UIDs you may ListGroups. However, if you already have group UIDs (therefore being able to call ExtractMembers), you would never ListGroups.

Still wrong.  Methods that only work because of the order in which they are called will break someday and it will be frustrating.